core/server/middleware/url-redirects.js

```diff
@@ -30,7 +32,11 @@ _private.getAdminRedirectUrl = function getAdminRedirectUrl(options) {
     debug('getAdminRedirectUrl', requestedHost, requestedUrl, adminHostWithProtocol);
 
     // CASE: we only redirect the admin access if `admin.url` is configured
-    if (adminHostWithProtocol !== utils.url.urlJoin(blogHostWithProtocol, 'ghost/')) {
+    // If url and admin.url are not equal AND the requested host does not match, redirect.
+    // The first condition is the most important, because it ensures that you have a custom admin url configured,
+    // because we don't force an admin redirect if you have a custom url configured, but no admin url.
+    if (adminHostWithoutProtocol !== utils.url.urlJoin(blogHostWithoutProtocol, 'ghost/') &&
+        adminHostWithoutProtocol !== utils.url.urlJoin(requestedHost, 'ghost/')) {
```

Comment:
Does this work with sub-directories? `requestedHost` won't include the sub-directory yet `(admin|blog)HostWithoutProtocol` will both include it unless I'm mistaken?

Reply:
Yeah good catch, will update my commit 👍